### PR TITLE
Add `count_group_by`, a function counting values of grouped rows

### PR DESCRIPTION
**Pull request by sehkone**
`Table::statistics` returns time series as well.

### CODE REVIEW

**Comment by msk** [comment]
This implementation is fine, but it is in a wrong layer. The purpose of this crate is to provide general mechanisms to store and manipulate structured data. However, the notion of `Statistics` doesn't really fit in the category of "general mechanisms"; rather, it is defined for a specific application.

How about providing more general functions, such as pandas' [groupby](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html), in this crate, and implementing application-specific features in the corresponding application using those general functions?

**Comment by sehkone** [comment]
I will do that.

**Comment by sehkone** [comment]
@msk , 

I didn't generalize this to support other action and types but just added a minimal interface.

File: src/table.rs [hunk 1]
```diff
@@ -202,6 +203,89 @@ impl Table {
...
+        by_interval: Option<u32>,
+        count_columns: &Arc<Vec<usize>>,
+    ) -> Vec<GroupCount> {
+        let rows_interval: Vec<GroupElement> = match column_types[by_column] {
+            ColumnType::DateTime => {
+                if let Some(by_interval) = by_interval {
+                    convert_time_intervals(
+                        self.columns
+                            .get(by_column)
+                            .expect("time column should exist"),
```
Comment (msk) [inline src/table.rs:222]: `count_group_by` is a public method, and thus we have to assume that its argument, `by_column`, may take any `usize` value.

Comment (sehkone) [reply src/table.rs:228]: Added validation of arguments, `by_column` and `count_columns`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +43,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
   column-oriented form.
 - Interface to read CSV data into `Table`.
 
+[Unreleased]: https://github.com/petabi/structured/compare/0.2.0...master
+[0.2.0]: https://github.com/petabi/structured/compare/0.1.1...0.2.0
```
[the remark this answers is not in the document]
Comment (sehkone) [reply CHANGELOG.md:47]: deleted.

File: src/stats.rs [hunk 3]
```diff
@@ -61,6 +71,19 @@ impl fmt::Display for Element {
     }
 }
 
+impl GroupElement {
+    pub fn cmp(&self, other: &GroupElement) -> std::cmp::Ordering {
```
[the remark this answers is not in the document]
Comment (sehkone) [reply src/stats.rs]: `PartialOrd` implemented now.

**Pull request merged**